instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -81,6 +85,13 @@ def inject_context(context, client_method)
           end
         end
 
+        def prepare_extract_context(context, client_method)
+          return unless client_method == SQS_RECEIVE_MESSAGE
+          return unless instrumentation_config[:extract_messaging_context]
+
+          context.params[:message_attribute_names] = ['All']
+        end
```

Comment:
This could be inlined with the check at L50 if you were so inclined; it's checking mostly the same things.